src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -490,3 +492,147 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
             f"('upgraded_from' = '{source}'"
             f" , '{table.UPGRADED_FROM_WS_PARAM}' = '{ws_id}');"
         )
+
+
+class ACLMigrator:
+    def __init__(
+        self,
+        table_crawler: TablesCrawler,
+        grant_crawler: GrantsCrawler,
+        ws: WorkspaceClient,
+        workspace_info: WorkspaceInfo,
+        backend: SqlBackend,
+        group_manager: GroupManager,
+        migration_status_refresher: MigrationStatusRefresher,
+        principal_grants: PrincipalACL,
+    ):
+        self._tc = table_crawler
+        self._gc = grant_crawler
+        self._backend = backend
+        self._workspace_info = workspace_info
+        self._workspace_name = workspace_info.current()
+        self._ws = ws
+        self._group = group_manager
+        self._migration_status_refresher = migration_status_refresher
+        self._principal_grants = principal_grants
+
+    def migrate_acls(
+        self,
+        *,
+        target_catalog: str | None = None,
+        legacy_table_acl: bool = True,
+        principal: bool = True,
+        hms_fed: bool = False,
+    ) -> None:
+        acl_strategy = []
+        if legacy_table_acl:
+            acl_strategy.append(AclMigrationWhat.LEGACY_TACL)
+        if principal:
+            acl_strategy.append(AclMigrationWhat.PRINCIPAL)
+        if acl_strategy is None:
+            return None
+        all_grants_to_migrate = self._gc.snapshot()
+        all_migrated_groups = self._group.snapshot()
+        all_principal_grants = self._principal_grants.get_interactive_cluster_grants()
+        tables = self._tc.snapshot()
+        tables_to_migrate = []
+        if not tables:
+            logger.info("No tables found to migrate")
+            return None
+        if hms_fed:
+            # if it is hms_fed acl migration, migrate all the acls for tables in the provided catalog
```

Comment:
this parent method is too long (~60 lines). create two new private methods - one for non-fed path and another for HMS fed.